README.md

```diff
@@ -1,23 +1,34 @@
+[![Package publish](https://github.com/microsoft/query-builder-for-digital-twins/actions/workflows/pipeline.yml/badge.svg)](https://github.com/microsoft/query-builder-for-digital-twins/actions/workflows/pipeline.yml)
+
+The Azure Digital Twins (ADT) QueryBuilder provides two C# based query builders, both are fluent query builders that help you build and query an Azure Digital Twin instance in an easy and predictable way with familiar C# based programming constructs.
+The first is in the namespace `QueryBuilder.Typed` and uses generics constrained by the BasicDigitalTwin class provided by the ADT Client SDK and provided Linq-like predicates to create queries.
+The second is in the namespace `QueryBuilder.Dynamic` and supports building queries in scenarios where the C# type of the model is not known, thus the syntax requires a bit more of verbosity. More often than not, you'll likely find that it to be a better experience to use the Typed query builder over the Dynamic one that has niche use cases. For a quick comparison between the Dynamic and Typed QueryBuilder read [here](#query-builder-comparison).
 
 Queries generated follows a grammar of custom SQL-like query language called [Azure Digital Twins query language](https://docs.microsoft.com/en-us/azure/digital-twins/concepts-query-language).
 
-## Pre-requisites
-There are some assumptions QueryBuilder is making so that it can be most useful (please refer to [ADT SDK](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/digitaltwins/Azure.DigitalTwins.Core#azure-iot-digital-twins-client-library-for-net) for more context):
+___
+
+## Prerequisites
+
+There are some assumptions the Typed QueryBuilder is making so that it can be most useful (please refer to [ADT SDK](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/digitaltwins/Azure.DigitalTwins.Core#azure-iot-digital-twins-client-library-for-net) for more context):
+
 - Digital Twin Definition Language (DTDL) models are represented in C# classes (C# models).
 - C# models inherit from BasicDigitalTwin class supplied by ADT SDK.
 - DTDL properties are mapped to C# properties in their corresponding models.
 - C# model properties have JsonPropertyName attributes mapping to their DTDL model property names, following the standards of ADT classes.
 - Enum values have EnumMember attributes mapping to the DTDL model enum values.
 - Models have references to their relationships and the relationship classes extend BasicRelationship class supplied by ADT SDK.
 - C# relationship classes set Name property at construction.
+- C# properties are attributed with the `[JsonProperty("propName")]` attribute from the `System.Text.Json` library. The value passed into the attribute corresponds with the property name in DTDL (particularly its casing).
```

Comment:
Weird, I read these multiple times because I thought to myself, "I'm sure he would've already included this point", and I added it because I thought I didn't see it. Thanks for catching this.